devito/finite_differences/differentiable.py

```diff
@@ -534,24 +530,63 @@ def expr(self):
     def dimensions(self):
         return self._dimensions
 
-    @property
-    def evaluate(self):
+    def _evaluate(self, **kwargs):
+        expr = self.expr._evaluate(**kwargs)
+
+        if not kwargs.get('expand', True):
+            return self.func(expr, self.dimensions)
+
         values = product(*[list(d.range) for d in self.dimensions])
         terms = []
         for i in values:
             mapper = dict(zip(self.dimensions, i))
-            terms.append(self.expr.xreplace(mapper))
+            terms.append(expr.xreplace(mapper))
         return sum(terms)
 
     @property
     def free_symbols(self):
         return super().free_symbols - set(self.dimensions)
 
 
+class Weights(Array):
+
+    """
+    The weights (or coefficients) of a finite-difference expansion.
+    """
+
+    def __init_finalize__(self, *args, **kwargs):
+        dimensions = as_tuple(kwargs.get('dimensions'))
+        weights = kwargs.get('initvalue')
+
+        assert len(dimensions) == 1
+        d = dimensions[0]
+        assert isinstance(d, StencilDimension) and d.symbolic_size == len(weights)
+        assert isinstance(weights, (list, tuple, np.ndarray))
+
+        kwargs['scope'] = 'static'
+
+        super().__init_finalize__(*args, **kwargs)
+
+    @property
+    def dimension(self):
+        return self.dimensions[0]
```

Comment:
it is always a single one, you may see that in the constructor we `assert len(dimensions) == 1`

If I don't add a `dimension` property, then _at each call site_ I have to perform an unpack such as `w.dimensions[0]` which also may confuse the reader because "what about the other dimensions"

A `dimension` property makes it much cleaner imho